src/api/data/tree/importPlan.ts

```diff
@@ -191,50 +191,38 @@ const replaceRefWithId =
 
 export const validatePlanContents =
   (logger: Logger) =>
```

Comment:
DataPlanPartFilesOnly shouldn't exist, it should be derived/inferred from the zod type.
and that schema should only have the real fields (not saveRefs/resolveRefs)

pro-tip: definitely read and understand the zod docs.  by default, zod.object will strip unknown props.  `.strictObject` will throw if any are present, and `.looseObject` will let them stay on the result.

so maybe warn about extra properties based on the `planContents` and then `.object` will strip them for you.

[FWIW, I'd also suggest using the actual CLI (oclif) warnings to help people migrate, rather than Logger which nobody will see]